### PR TITLE
DRILL-7904: Update to 30-jre Guava version

### PR DESCRIPTION
**Pull request by vdiravka**
# [DRILL-7904](https://issues.apache.org/jira/browse/DRILL-7904): Update to 30-jre Guava version

## Description

* Update guava version to the newest one (30.1.1-jre)
* Drop <shaded.guava.version> at all

## Documentation
NA

## Testing
@vvysotskyi Since the original Guava shade was performed by you please advice what is needed to be tested before merging?

### CODE REVIEW

**Comment by vvysotskyi** [comment]
@vdiravka, thanks for the pull request. 
Could you please create two commits: one commit with all manual changes and another one with changes related to renaming packages?

Regarding checks, the main one is clean tests run.

After that, please check how Drill will work with hive, hbase, etc.
Please make sure that Drill JDBC Driver is not broken, try to connect to Drill Through SQuirreL and DBeaver, try setting up custom authentication, and submit different queries.
Also, please make sure that Drill is starting and running in distributed mode, Web-UI is functioning correctly.

Also, Drill has some code related to patching guava classes, could you please check whether it is still needed and if yes that it wasn't broken by the update?

File: distribution/src/assemble/component.xml
Comment (vvysotskyi) [inline distribution/src/assemble/component.xml:95]: Please remove the whole `dependencySet` block, since it is not needed without `includes`.

File: docs/dev/ArtidfactsPublishing.md
Comment (vvysotskyi) [inline docs/dev/ArtidfactsPublishing.md:1]: Please remove whole the file, since except for shaded guava, there is no need anymore for publishing libraries manual.

File: pom.xml
Comment (vvysotskyi) [inline pom.xml:1051]: Please remove the `version` line, version should be obtained from the dependencies management block.

**Review by vvysotskyi — CHANGES_REQUESTED** [review]
Please also remove `GuavaUtils` and its usage.

And in `exec/jdbc-all/pom.xml` please remove all entries with `org/apache/drill/shaded/guava...`. Also, since we don't ship shaded classes, the jar size limit may be decreased.

File: docs/dev/ArtidfactsPublishing.md
Comment (vdiravka) [reply docs/dev/ArtidfactsPublishing.md:1]: I thought to it too. But this info can be useful for any other library shade, can't it be?

File: distribution/src/assemble/component.xml
Comment (vdiravka) [reply distribution/src/assemble/component.xml:95]: right, thanks!

**Comment by vdiravka** [comment]
@vvysotskyi Looks like unshade is not possible right now. We can update main guava lib version. As a next step we can consider update Drill Guava shade version to the latest one.

**Comment by vvysotskyi** [comment]
@vdiravka, it will be simpler to update the shaded version at first, since it is used by Drill only, so no 3rd-party libraries will be broken (and fewer checks should be done). Could you please prepare the list of unresolved issues you faced when you tried replacing the shaded version and put it in the Jira ticket description, so it will be simpler to define the scope of work to do when we will try removing it again?

**Comment by vdiravka** [comment]
ok. I will update publish Drill Shaded Guava `30.1.1-jre` version to `repository.apache.org` and update current `28.2-jre` in scope of this PR

**Comment by laurentgo** [comment]
@vdiravka @vvysotskyi @cgivre it looks like the shaded guava was published and the build with it passed. Any objection getting this change merged?

**Comment by vvysotskyi** [comment]
@laurentgo, looks like changes in this PR just update the non-shaded guava version. But since it is used by lot 3rd party libraries, all checks from this comment https://github.com/apache/drill/pull/2202#issuecomment-825797242 should be produced before merging the PR to avoid regressions.

**Comment by laurentgo** [comment]
Is there an imperative for this change to be merged in 1.19 then, or can it be part of the next version?

**Comment by vvysotskyi** [comment]
I'm fine with any of the options, but let's wait for the response from @cgivre and @vdiravka.

**Comment by cgivre** [comment]
> I'm fine with any of the options, but let's wait for the response from @cgivre and @vdiravka.

+1 from me, but I'd defer to @vdiravka.

**Comment by vdiravka** [comment]
I started the process of updating Drill shaded Guava. But it will not be included into the 1.19.0.
And since we start discussing let's merge it.

**Comment by vvysotskyi** [comment]
@vdiravka, could you please confirm that all the checks from the comment above were produced?

**Pull request merged**
